src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -359,147 +360,180 @@ async Task<JObject> ResolveAsInstanceMember(ArraySegment<string> parts, JObject
+                        else
+                        {
+                            return (JObject)(((JArray)rootObject["value"]).FirstOrDefault(x => x["name"].Value<string>() == elementIdxInfo.ElementIdxStr)["value"]);
+                        }
+                    case "object":
+                        if (elementIdxInfo.IsMultidimensional)
+                            throw new InvalidOperationException($"Cannot apply indexing with [,] to an object of type '{type}'");
+                        // ToDo: try to use the get_Item for string as well
+                        if (type == "string")
+                        {
+                            var eaExpressionFormatted = elementAccessStrExpression.Replace('.', '_'); // instance_str
+                            variableDefinitions.Add(ExpressionEvaluator.ConvertJSToCSharpLocalVariableAssignment(eaExpressionFormatted, rootObject));
+                            var eaFormatted = elementAccessStr.Replace('.', '_'); // instance_str[1]
+                            return await ExpressionEvaluator.EvaluateSimpleExpression(this, eaFormatted, elementAccessStr, variableDefinitions, logger, token);
+                        }
+                        var typeIds = await context.SdbAgent.GetTypeIdsForObject(objectId.Value, true, token);
+                        int[] methodIds = await context.SdbAgent.GetMethodIdsByName(typeIds[0], "get_Item", BindingFlags.Default, token);
+                        if (methodIds == null)
+                            throw new InvalidOperationException($"Type '{rootObject?["className"]?.Value<string>()}' cannot be indexed.");
+
+                        // get_Item should not have an overload, but if user defined it, take the default one: with one param (key)
+                        int allowedParamCnt = 1;
+                        int toArrayId = methodIds[0];
+                        // ToDo: optimize the loop by choosing the right method at once without trying out them all
+                        for (int i = 0; i < methodIds.Length; i++)
+                        {
+                            MethodInfoWithDebugInformation methodInfo = await context.SdbAgent.GetMethodInfo(methodIds[i], token);
+                            ParameterInfo[] paramInfo = methodInfo.GetParametersInfo();
+
+                            if (paramInfo.Length == allowedParamCnt)
                             {
-                                var arg = elementAccess.ArgumentList.Arguments[i];
-                                if (i != 0)
+                                try
                                 {
-                                    elementIdxStr += ", ";
-                                    multiDimensionalArray = true;
+                                    if (indexObject is null && elementIdxInfo.IndexingExpression is null)
+                                        throw new InternalErrorException($"Unable to write index parameter to invoke the method in the runtime.");
```

Comment:
Does this need to be in the loop? This can be checked earlier, maybe right before
https://github.com/dotnet/runtime/blob/496678ac64a5bee4b5d8d0da937e8ab3e337f5ad/src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs#L420-L421

Reply:
True